library/goptions.ml

```diff
@@ -422,9 +422,16 @@ let set_option_value ?(locality = OptDefault) check_and_cast key v =
   let opt = try Some (get_option key) with Not_found -> None in
   match opt with
   | None -> warn_unknown_option key
-  | Some (depr, (read,write,append)) ->
+  | Some (depr, (read,write,_)) ->
     write locality (check_and_cast v (read ()))
 
+let append_option_value ?(locality = OptDefault) check_and_cast key v =
```

Comment:
Doesn't this duplicate `set_string_option_append_value_gen`? (if you want to rename it that's fine, I don't even know why it's called _gen)